Library/Homebrew/formula_auditor.rb

```diff
@@ -314,15 +314,38 @@ def audit_deps
     end
 
     def audit_conflicts
-      formula.conflicts.each do |c|
-        Formulary.factory(c.name)
+      formula.conflicts.each do |conflict|
+        conflicting_formula = Formulary.factory(conflict.name)
+        next if formula.tap != conflicting_formula.tap
+
+        problem "Formula should not conflict with itself" if formula == conflicting_formula
+
+        tap = formula.tap
```

Comment:
Not blocking but I think this could be pulled out of the `#each` block (but doesn't have to be)